src/NHibernate/Driver/NDataReader.cs

```diff
@@ -549,6 +552,21 @@ internal static NResult Create(DbDataReader reader, bool isMidstream)
 				return result;
 			}
```

Comment:
Here we want to delay throwing NotSupportedException until when someone actually calls `GetSchemaTable`.